sdk/src/index.ts

```diff
@@ -6,13 +6,32 @@ import { DeviceBase, ScryptedInterfaceDescriptors, ScryptedInterfaceProperty, TY
  * @category Core Reference
  */
 export class ScryptedDeviceBase extends DeviceBase {
-  private _storage: Storage;
-  private _log: Logger;
-  private _console: Console;
-  private _deviceState: DeviceState;
+  private _storage: Storage | undefined;
+  private _log: Logger | undefined;
+  private _console: Console | undefined;
+  private _deviceState: DeviceState | undefined;
 
   constructor(public readonly nativeId?: string) {
     super();
+
+    addScryptedInterfaceProperties({
+      thisObject: this,
```

Comment:
This a significant a behavioral change. It's running on every constructor and modifying the `this` object whereas prior it was modifying the class prototypes. While it may be valid, the change gives me the willies, and I think should mimic the existing behavior.

Reply:
Fair enough. I restored the original self executing function that modified the class prototypes. https://github.com/koush/scrypted/pull/1308/commits/576b59260d97e3e3ffb6cef149474693d60aece9

However to make TypeScript happy, I had to make the class `_deviceState` public because we're accessing it from outside the class.

https://github.com/koush/scrypted/pull/1308/commits/576b59260d97e3e3ffb6cef149474693d60aece9#diff-8db04fc25758a2af0f10ddcc8941574a24d2cfe1a3dc65f87204b1ca995145cbR12

This isn't perfect but I couldn't think of another solution.

Reply:
cast it as `any` and then grab it. 🤷🏽

Reply:
actually ya ts-ignore might work too

Reply:
I tested unfortunately `ts-ignore` won't work for rename refactor either because TypeScript doesn't actually link the `_deviceState` to the property at all.

Reply:
If I have it as `public _deviceState` then the rename refactor works.

Reply:
yeah I think changing the signature back to `state: any` is best, the API needs to hide that.

Reply:
Although could argue `ts-ignore` is slightly better than `thisObject: any` since neither would allow for rename refactor but we get type-safety when calling `deviceBase._lazyLoadDeviceState();`

Reply:
I added ts-ignore, let me know if you disagree https://github.com/koush/scrypted/pull/1308/commits/907b4c2976a7b5bd4a5794386a9c52ab1c222ecf